specs/DisableNavigatingBackAndForward.md

```diff
@@ -0,0 +1,124 @@
+decide whether to suppress. 
+
+This should be solvable in a generic way. However, this feature hasn’t been released yet, and it might
+be better if we could provide a simpler and more direct way. 
+
+Therefore, our job is to provide a mechanism for developers to disable navigating back and forward
+without excessive effort.
+
+
+# Examples
+#### Win32 C++
+
+##### Use `ICoreWebView2NavigationStartingEventArgs3` in `add_NavigationStarting`
+
+```c++
+//! [NavigationStarting]
+// Register a handler for the NavigationStarting event.
+// This handler will check the navigation status, and if the navigation is
+// `GoBack` or `GoForward`, it will be canceled.
+CHECK_FAILURE(m_webView->add_NavigationStarting(
+    Callback<ICoreWebView2NavigationStartingEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2NavigationStartingEventArgs* args)
+        -> HRESULT {
+            wil::com_ptr<ICoreWebView2NavigationStartingEventArgs3> args3;
+            if (SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args3))))
+            {
+                COREWEBVIEW2_NAVIGATION_KIND kind;
+                CHECK_FAILURE(args3->get_NavigationKind(&kind));
+                // disable navigation if it is back/forward
+                if (kind == COREWEBVIEW2_NAVIGATION_KIND_BACKORFORWARD)
+                {
+                     CHECK_FAILURE(args->put_Cancel(true));
+                }
+            }
+            return S_OK;
+        })
+    .Get(),
+    &m_navigationStartingToken));
+//! [NavigationStarting]
+```
+
+#### .NET and WinRT
+
+#### Use `CoreWebView2NavigationStartingEventArgs` in `NavigationStarting`
+
+```c#
+// Register a handler for the NavigationStarting event.
+// This handler will check the navigation status, and if the navigation is
+// `GoBack` or `GoForward`, it will be canceled.
+void WebView_NavigationStarting(object sender, CoreWebView2NavigationStartingEventArgs e)
+{
+    if (e.NavigationKind == CoreWebView2NavigationKind.BackOrForward)
+    {
+        e.Cancel = true;
+    }
+}
+```
+
+# API Details
+#### Win32 C++
+
+```c++
+// Enums and structs
+[v1_enum]
+typedef enum COREWEBVIEW2_NAVIGATION_KIND {
+  /// A navigation caused by CoreWebView2.Reload(), location.reload(), the end user using F5 or other UX, or other reload mechanisms to reload the current document without modifying the navigation history.
+  COREWEBVIEW2_NAVIGATION_KIND_RELOAD,
+  /// A navigation back or forward to a different entry in the session navigation history. For example via CoreWebView2.Back(), location.back(), the end user pressing Alt+Left or other UX, or other mechanisms to navigate forward or backward in the current session navigation history.
+  COREWEBVIEW2_NAVIGATION_KIND_BACKORFORWARD,
+  /// A navigation to a different document. This can be caused by CoreWebView2.Navigate(), window.location.href = '...', or other WebView2 or DOM APIs that navigate to a specific URI.
+  COREWEBVIEW2_NAVIGATION_KIND_DIFFERENT,
```

Comment:
Does "different" always mean "Different Document" as mentioned in the comment?  If so, I think we should fully specify that in the enum value name.  By itself, "Different" seems functionally similar to "Other" catch-all which we wanted to avoid.

Reply:
Yes. It means `Different Document`. I just copy that from the `NavigationType`, and I will use `DifferentDocument` instead.